cmd/templatizer/app/flags.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// Options represent configurable parameters for jaeger-templatizer
+type Options struct {
+	Mapping   string
+	EsVersion int64
+	Shards    int64
+	Replicas  int64
+	EsPrefix  string
+	UseILM    string // using string as util is being used in python and using bool leads to type issues.
+}
+
+const (
+	mappingFlag   = "mapping"
+	esVersionFlag = "esVersion"
+	shardsFlag    = "shards"
+	replicasFlag  = "replicas"
+	esPrefixFlag  = "esPrefix"
+	useILMFlag    = "useILM"
+)
+
+// AddFlags adds flags for templatizer main program
+func (o *Options) AddFlags(command *cobra.Command) {
+	command.Flags().StringVarP(
+		&o.Mapping,
+		mappingFlag,
+		"m",
+		"",
+		"The index mapping the template will be applied to. Pass either jaeger-span or jaeger-service")
+	command.Flags().Int64VarP(
+		&o.EsVersion,
+		esVersionFlag,
+		"v",
+		7,
+		"The major Elasticsearch version")
+	command.Flags().Int64VarP(
+		&o.Shards,
+		shardsFlag,
+		"s",
+		5,
+		"The number of shards per index in Elasticsearch")
+	command.Flags().Int64VarP(
+		&o.Replicas,
+		replicasFlag,
+		"r",
+		1,
+		"The number of replicas per index in Elasticsearch")
+	command.Flags().StringVarP(
+		&o.EsPrefix,
+		esPrefixFlag,
+		"p",
+		"",
+		"Specifies index prefix")
+	command.Flags().StringVarP(
```

Comment:
when using stdlib flags, there is `flags.Bool`

Reply:
Cobra does have BoolVarP support, this was intentionally made string - as when I was calling this tool from python, if I used Bool flag, I was getting type conflicts.

Reply:
Is it just because Booleans in Python are capitalized?

Reply:
Yes. I guess that was the case. I dont exactly remember now though.